README.md

```diff
@@ -1,51 +1,88 @@
 
-This is a library for writing plugins for the [snap](https://github.com/intelsdi-x/snap) telemetry framework. The goal for this library is to make it super simple to write a plugin.
+This is a library for writing plugins for the [Snap](https://github.com/intelsdi-x/snap) telemetry framework. The goal for this library is to make it super simple to write a plugin.
 
-# Writing a plugin
+Snap itself runs as a master daemon with the core functionality that may load and unload plugin processes via either CLI or HTTP APIs.
 
-In order to write a plugin for snap, it is necessary to define some methods to satisfy the appropriate interfaces. The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. The interfaces are:
+A Snap plugin is a program, or a set of functions or services, written in Go or any language; that may seamlessly integrate with Snap as executables.
 
+Communication between Snap and plugins uses the GRPC protocol. GRPC uses HTTP2, which improves performance versus HTTP1, and uses [protocol buffers](https://developers.google.com/protocol-buffers/) for simplicity, performance, and smaller memory size.
 
-```go
-type Plugin interface {
-	GetConfigPolicy() (ConfigPolicy, error)
-}
+Before starting writing Snap plugins, check out the Plugin Catalog to see if any suit your needs. If not, you need to reference the plugin packages that defines the type of structures and interfaces inside snap and then write plugin endpoints to implement the defined interfaces.
 
-// Collector is a plugin which is the source of new data in the Snap pipeline.
-type Collector interface {
-	Plugin
 
-	GetMetricTypes(Config) ([]Metric, error)
-	CollectMetrics([]Metric) ([]Metric, error)
-}
+## Snap Plugin Go Library Examples
+You will find [example plugins](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/examples) that cover the basics for writing collector, processor, and publisher plugins in teh examples folder.
 
-// Processor is a plugin which filters, agregates, or decorates data in the
-// Snap pipeline.
-type Processor interface {
-	Plugin
+## Snap Plugin Go Library Implementaion Files
+You will find all the library packages in the [v1/plugin folder](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/v1/plugin).
+The [rpc folder](https://github.com/intelsdi-x/snap-plugin-lib-go/tree/master/v1/plugin/rpc) contents are auto-generated and should not be modified by anyone.
 
-	Process([]Metric, Config) ([]Metric, error)
-}
 
-// Publisher is a sink in the Snap pipeline.  It publishes data into another
-// System, completing a Workflow path.
-type Publisher interface {
-	Plugin
+## Plugin Naming, Files, and Directory
+For your plugin, create a new repository and name your plugin project using the following format:
 
-	Publish([]Metric, Config) error
-}
+>snap-plugin-[plugin-type]-[plugin-name]
+
+There are three plugin types supported by Snap: collector, processor, publisher
```

Comment:
is this information right to have in the README or under each example? Mapping out the developer's path to this page makes me think that the README should be a brief intro, links to how-tos for each type, overview of what/why, links to where to get involved and that's it. Leaning toward limiting repetitive information in the README, though I'm concerned with duplicate in each plugin how-to as well. It's your call whether the README has that core introduction or each folder has a complete tutorial.